### PR TITLE
chore: initial project structure [part 1/5]

### PR DESCRIPTION
**Pull request by andreabadesso**
### Motivation

We should create the initial project structure, including all dev dependencies

### Acceptance Criteria
- We should have a `flake.nix` defining all system dependencies we need and it should be loaded by `nix-direnv` automatically (through the .envrc)
- The project should be a `yarn workspaces` project, with a single package, the `hathor-rpc-handler`
- The `hathor-rpc-handler` package should include `eslint`, `tseslint` and `typescript` and use fixed versions
- We should use the recommended lint configs from `eslint` and `tseslint`

### Checklist
- [X] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged
- [X] Make sure either the unit tests and/or the QA tests are capable of testing the new features
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/release_pr_template.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+Link here all the PRs that are included in this release
+
+# Release-candidates
+
+Link here the release-candidates that were deployed as part of this release
+
+# Checklist
+
+- [ ] I've read and followed the release process described in https://github.com/HathorNetwork/ops-tools/blob/master/docs/release-guides/wallet-service.md#stable-release
```
Comment (pedroferreira1) [inline .github/PULL_REQUEST_TEMPLATE/release_pr_template.md:11]: Link is from wallet service release process

File: package.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "hathor-rpc-lib",
+  "private": true,
```
Comment (pedroferreira1) [inline package.json:3]: Why is it private?

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{
+  description = "virtual environments";
+
+  inputs = {
+    devshell.url = "github:numtide/devshell";
+    flake-utils.url = "github:numtide/flake-utils";
+    unstableNixPkgs.url = "nixpkgs/nixos-unstable";
```
Comment (pedroferreira1) [inline flake.nix:7]: What's this? Why is it unstable?

File: packages/hathor-rpc-handler/eslint.config.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+const eslint = require('@eslint/js');
+const tseslint = require('typescript-eslint');
+
+module.exports = tseslint.config(
+  eslint.configs.recommended,
+  ...tseslint.configs.recommended,
+  {
+    ignores: [
+      'eslint.config.js',
```
Comment (pedroferreira1) [inline packages/hathor-rpc-handler/eslint.config.js:11]: Why do you need to ignore this?

File: package.json (hunk 2 above)
Comment (andreabadesso) [reply package.json:3]: This is the root package, which will never be published, so it's private

The actual packages that will be build are inside `packages/`

File: .gitattributes [hunk 5]
```diff
@@ -0,0 +1,4 @@
+/.yarn/**            linguist-vendored
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply .gitattributes:1]: Yarn created these automatically, they help git, github or whatever git client that is reading them understand what these files are (like not showing a diff when the file is binary

Kind of pointless though as we are adding `.yarn` to `.gitignore`, I'll delete this file

File: .github/PULL_REQUEST_TEMPLATE/release_pr_template.md (hunk 1 above)
Comment (andreabadesso) [reply .github/PULL_REQUEST_TEMPLATE/release_pr_template.md]: Removed in 50213095c637848b9edcd20ec49af774453aad4d

File: flake.nix (hunk 3 above)
Comment (andreabadesso) [reply flake.nix]: I was using yarn from the unstable nix channel as the latest version didn't have yarn v2 available

It's now available, so we don't need to fetch it from the unstable channel

updated in f5fa3d18ad6aa71a3236dee14ac6aeb8a4a2f6fd

File: packages/hathor-rpc-handler/eslint.config.js (hunk 4 above)
Comment (andreabadesso) [reply packages/hathor-rpc-handler/eslint.config.js]: Fixed in 61983cabfb8c4b125baca60a8827dcd54ad4e41a

File: packages/hathor-rpc-handler/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,2 @@
+export default () => {
```
Comment (tuliomir) [inline packages/hathor-rpc-handler/src/index.ts:1]: question: Should we have the Hathor header here?

**Review by pedroferreira1 — APPROVED** [review]

File: packages/hathor-rpc-handler/src/index.ts (hunk 6 above)
Comment (andreabadesso) [reply packages/hathor-rpc-handler/src/index.ts:8]: Done! Thanks

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,15 @@
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply .gitignore]: Yarn added those automatically, I don't like them at all..

[Their justification](https://yarnpkg.com/features/caching#zero-installs) is that they want everything to be contained in the project, so all devs share the exactly same environment

I'm using nix for this exact purpose and I think it's way superior... 

Updated gitignore to remove those

Thanks

**Review by tuliomir — APPROVED** [review]

**Pull request merged**